File: Backend/TailorApp.API/Models/NameValue.cs
namespace TailorApp.API.Models;

public class NameValue
{
    public int NameValueId { get; set; }
    public string Category { get; set; } = string.Empty;
    // e.g. Unit, GarmentType, FabricType, OrderStatus

    public string Value { get; set; } = string.Empty;
    // e.g. inch, cm, Shirt, Pant, Cotton

    public string? Label { get; set; }
    // Display label (optional, falls back to Value)

    public int SortOrder { get; set; } = 0;
    public bool IsActive { get; set; } = true;
}

File: Frontend/tailor-app-ui/src/app/models/namevalue.ts
export interface NameValue {
  nameValueId: number;
  category: string;
  value: string;
  label: string;
  sortOrder: number;
}

export const NV_CATEGORIES = {
  UNIT:         'Unit',
  GARMENT_TYPE: 'GarmentType',
  FABRIC_TYPE:  'FabricType'
};